src/stores/blockstore.js

```diff
@@ -52,19 +52,22 @@ exports.setUp = (basePath, BlobStore, locks) => {
       }
 
       const p = multihashToPath(key, extension)
+      const deferred = defer()
+
+      lock(p, (release) => {
+        const ext = extension === 'data' ? 'protobuf' : extension
+        pull(
+          store.read(p),
+          pull.collect(release((err, data) => {
+            if (err) return deferred.abort(err)
```

Comment:
I do remember having a chat about this a long time ago and we agreed with curlys, AFAIK that decision hasn't changed